From eb674b1d4101bab3e5d2851d0cd2d75ee4f7e768 Mon Sep 17 00:00:00 2001
From: 5656565566 <contact@home56.top>
Date: Wed, 12 Aug 2026 23:55:47 +0800
Subject: [PATCH] abi-dumps: add Floral NativeBridge private symbols

Track the process-context setter and the hybrid loader opt-in query exported by libnativebridge.

Change-Id: Ie35541004a047dddb59306091db89b85dc83617e
---
 31/64/arm64_armv8-a/source-based/libnativebridge.so.lsdump  | 6 ++++++
 .../arm_armv7-a-neon/source-based/libnativebridge.so.lsdump | 6 ++++++
 31/64/arm_armv8-a/source-based/libnativebridge.so.lsdump    | 6 ++++++
 31/64/x86/source-based/libnativebridge.so.lsdump            | 6 ++++++
 31/64/x86_64/source-based/libnativebridge.so.lsdump         | 6 ++++++
 31/64/x86_x86_64/source-based/libnativebridge.so.lsdump     | 6 ++++++
 6 files changed, 36 insertions(+)

diff --git a/31/64/arm64_armv8-a/source-based/libnativebridge.so.lsdump b/31/64/arm64_armv8-a/source-based/libnativebridge.so.lsdump
index 63981bb..796bb73 100644
--- a/31/64/arm64_armv8-a/source-based/libnativebridge.so.lsdump
+++ b/31/64/arm64_armv8-a/source-based/libnativebridge.so.lsdump
@@ -135,11 +135,17 @@
   {
    "name" : "NativeBridgeInitialized"
   },
+  {
+   "name" : "NativeBridgeUseHybridLoader"
+  },
   {
    "name" : "NeedsNativeBridge"
   },
   {
    "name" : "PreInitializeNativeBridge"
+  },
+  {
+   "name" : "SetFloralNativeBridgeProcessContext"
   }
  ],
  "elf_objects" : [],
diff --git a/31/64/arm_armv7-a-neon/source-based/libnativebridge.so.lsdump b/31/64/arm_armv7-a-neon/source-based/libnativebridge.so.lsdump
index 1ec9f55..eb485ec 100644
--- a/31/64/arm_armv7-a-neon/source-based/libnativebridge.so.lsdump
+++ b/31/64/arm_armv7-a-neon/source-based/libnativebridge.so.lsdump
@@ -135,11 +135,17 @@
   {
    "name" : "NativeBridgeInitialized"
   },
+  {
+   "name" : "NativeBridgeUseHybridLoader"
+  },
   {
    "name" : "NeedsNativeBridge"
   },
   {
    "name" : "PreInitializeNativeBridge"
+  },
+  {
+   "name" : "SetFloralNativeBridgeProcessContext"
   }
  ],
  "elf_objects" : [],
diff --git a/31/64/arm_armv8-a/source-based/libnativebridge.so.lsdump b/31/64/arm_armv8-a/source-based/libnativebridge.so.lsdump
index 1ec9f55..eb485ec 100644
--- a/31/64/arm_armv8-a/source-based/libnativebridge.so.lsdump
+++ b/31/64/arm_armv8-a/source-based/libnativebridge.so.lsdump
@@ -135,11 +135,17 @@
   {
    "name" : "NativeBridgeInitialized"
   },
+  {
+   "name" : "NativeBridgeUseHybridLoader"
+  },
   {
    "name" : "NeedsNativeBridge"
   },
   {
    "name" : "PreInitializeNativeBridge"
+  },
+  {
+   "name" : "SetFloralNativeBridgeProcessContext"
   }
  ],
  "elf_objects" : [],
diff --git a/31/64/x86/source-based/libnativebridge.so.lsdump b/31/64/x86/source-based/libnativebridge.so.lsdump
index e4443f1..1af0d0f 100644
--- a/31/64/x86/source-based/libnativebridge.so.lsdump
+++ b/31/64/x86/source-based/libnativebridge.so.lsdump
@@ -134,11 +134,17 @@
   {
    "name" : "NativeBridgeInitialized"
   },
+  {
+   "name" : "NativeBridgeUseHybridLoader"
+  },
   {
    "name" : "NeedsNativeBridge"
   },
   {
    "name" : "PreInitializeNativeBridge"
+  },
+  {
+   "name" : "SetFloralNativeBridgeProcessContext"
   }
  ],
  "elf_objects" : [],
diff --git a/31/64/x86_64/source-based/libnativebridge.so.lsdump b/31/64/x86_64/source-based/libnativebridge.so.lsdump
index e9b8708..6185224 100644
--- a/31/64/x86_64/source-based/libnativebridge.so.lsdump
+++ b/31/64/x86_64/source-based/libnativebridge.so.lsdump
@@ -134,11 +134,17 @@
   {
    "name" : "NativeBridgeInitialized"
   },
+  {
+   "name" : "NativeBridgeUseHybridLoader"
+  },
   {
    "name" : "NeedsNativeBridge"
   },
   {
    "name" : "PreInitializeNativeBridge"
+  },
+  {
+   "name" : "SetFloralNativeBridgeProcessContext"
   }
  ],
  "elf_objects" : [],
diff --git a/31/64/x86_x86_64/source-based/libnativebridge.so.lsdump b/31/64/x86_x86_64/source-based/libnativebridge.so.lsdump
index e4443f1..1af0d0f 100644
--- a/31/64/x86_x86_64/source-based/libnativebridge.so.lsdump
+++ b/31/64/x86_x86_64/source-based/libnativebridge.so.lsdump
@@ -134,11 +134,17 @@
   {
    "name" : "NativeBridgeInitialized"
   },
+  {
+   "name" : "NativeBridgeUseHybridLoader"
+  },
   {
    "name" : "NeedsNativeBridge"
   },
   {
    "name" : "PreInitializeNativeBridge"
+  },
+  {
+   "name" : "SetFloralNativeBridgeProcessContext"
   }
  ],
  "elf_objects" : [],
-- 
2.47.3

